components/camel-weka/src/main/java/org/apache/camel/component/weka/WekaConsumer.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.weka;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.component.weka.WekaConfiguration.Command;
+import org.apache.camel.support.DefaultConsumer;
+
+import io.nessus.weka.Dataset;
+
+public class WekaConsumer extends DefaultConsumer {
+
+    public WekaConsumer(WekaEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+    }
+
+    @Override
+    public WekaEndpoint getEndpoint() {
+        return (WekaEndpoint)super.getEndpoint();
+    }
+
+    public WekaConfiguration getConfiguration() {
+        return getEndpoint().getConfiguration();
+    }
+    
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+        
+        Thread thread = new Thread(new Runnable() {
```

Comment:
Its preferred to create thread via Camel. See CamelContext ExecutorServiceStrategy which has APIs for creating threads. Also you should stop the thread in doStop.